### PR TITLE
Jira confluence PAT

### PR DESCRIPTION
**Pull request by babenek**
## Description

- Add Jira/confluence PAT sample and pattern
- Applied entropy check for small random data in the token type
[entropy.ods](https://github.com/Samsung/CredSweeper/files/11581984/entropy.ods) - experements for the function
[entropy.xls](https://github.com/Samsung/CredSweeper/files/11589801/entropy.xls) -xls

- added min_length for each rule to spedup common process.

## How has this been tested?

- [x] UnitTest
- [x] Fuzzing coverage will be fixed in next PR #340

### CODE REVIEW

**Review by Yullia — APPROVED** [review]

**Review by Yullia — APPROVED** [review]

File: credsweeper/utils/util.py [hunk 1]
```diff
@@ -88,13 +88,82 @@ def get_shannon_entropy(data: str, iterator: str) -> float:
...
+    def get_min_data_entropy(x: int) -> float:
+        """Returns minimal entropy for size of random data. Precalculated data is applied for speedup"""
+        if x in Util.MIN_DATA_ENTROPY:
+            y = Util.MIN_DATA_ENTROPY[x]
+        elif 8 < x < 64:
+            # approximated for range 12 - 64
+            _x = x - 8
+            y = ((0.000016617804 * _x - 0.002695077) * _x + 0.170393) * _x + 0.4
+        elif 64 < x:
+            # logarithm base 2 - slowest but precisely
```
Comment (kmnls) [inline credsweeper/utils/util.py:118]:
```suggestion
            # logarithm base 2 - slowest but precise
```

File: credsweeper/utils/util.py [hunk 2]
```diff
@@ -88,13 +88,82 @@ def get_shannon_entropy(data: str, iterator: str) -> float:
...
+            _x = x - 8
+            y = 1.581026279659 * math.log2(_x) - 1.90156
+        else:
+            # less or equal 8 bytes might be 0 entropy
```
Comment (kmnls) [inline credsweeper/utils/util.py:122]:
```suggestion
            # less or equal to 8 bytes might have 0 entropy
```

File: credsweeper/utils/util.py [hunk 3]
```diff
@@ -88,13 +88,82 @@ def get_shannon_entropy(data: str, iterator: str) -> float:
...
+            y = 0
+        return y
+
+    @staticmethod
+    def ascii_entropy_small_data_test(data: bytes) -> bool:
+        """
+        Tests small data sequence (<256) for random data with ascii test and shannon entropy
```
Comment (kmnls) [inline credsweeper/utils/util.py:129]:
```suggestion
        Tests small data sequence (<256) for data randomness by testing for ascii and shannon entropy
```

File: credsweeper/utils/util.py [hunk 4]
```diff
@@ -88,13 +88,82 @@ def get_shannon_entropy(data: str, iterator: str) -> float:
...
+        Returns True when data is an ASCII symbols or have small entropy
+        """
+        if not data:
+            return True
+        data_len = len(data)
+        if 9 > data_len:
+            # even random data may have 0 entropy with length 8 bytes and less
```
Comment (kmnls) [inline credsweeper/utils/util.py:136]:
```suggestion
            # even random data may have 0 entropy for lengths of 8 bytes and less
```

File: tests/filters/test_value_structured_token_check.py [hunk 5]
```diff
@@ -3,33 +3,35 @@
...
 from credsweeper.filters import ValueStructuredTokenCheck
 from tests.filters.conftest import LINE_VALUE_PATTERN
 from tests.test_utils.dummy_line_data import get_line_data
 
 
 class TestValueStructuredTokenCheck:
 
-    @pytest.mark.parametrize("line", ["12345:asbdsa:28yd"])
+    @pytest.mark.parametrize("line", ["12345:«VÏel\tSpÁß¡»q¾ä×^0÷ÕJür°eÆ\r"])
     def test_value_structured_token_check_p(self, file_path: pytest.fixture, line: str) -> None:
```
Comment (kmnls) [inline tests/filters/test_value_structured_token_check.py:15]: Why this test is marked _p and following _n? It seems that their conditions are inverted

Comment (babenek) [reply tests/filters/test_value_structured_token_check.py:15]: Order P, N - is a requirement.
Filter returns true if the value must be filtered. False - when it should be kept (positive value for rule).

File: tests/utils/test_util.py [hunk 6]
```diff
@@ -285,6 +287,31 @@ def test_is_binary_n(self):
         self.assertFalse(Util.is_elf(AZ_STRING.encode("utf_16_le")))
         self.assertFalse(Util.is_elf(AZ_STRING.encode("utf_16_be")))
 
+    def test_get_shannon_data_entropy_p(self):
+        self.assertTrue(Util.ascii_entropy_small_data_test(b''))
+        self.assertTrue(Util.ascii_entropy_small_data_test(AZ_DATA))
+        # remove all spaces to make a variable name
+        az_data = AZ_DATA.replace(b' ', b'')  # 35 bytes
+        self.assertTrue(Util.ascii_entropy_small_data_test(az_data))
+        hangul_pangram_data = "키스의 고유 조건은 입술 끼리 만나야 하고 특별한 기술은 필요치 않다.".encode(UTF_8)
```
Comment (csh519) [inline tests/utils/test_util.py:296]: :rofl:

File: credsweeper/utils/util.py [hunk 7]
```diff
@@ -88,13 +88,82 @@ def get_shannon_entropy(data: str, iterator: str) -> float:
...
+            # logarithm base 2 - slow, but precise
+            _x = x - 8
+            y = 1.581026279659 * math.log2(_x) - 1.90156
+        else:
+            # less or equal to 8 bytes might have 0 entropy
+            y = 0
+        return y
+
+    @staticmethod
+    def ascii_entropy_small_data_test(data: bytes) -> bool:
```
Comment (csh519) [inline credsweeper/utils/util.py:127]: IMO this function name would be good to change with the format which start with `is_`.
As you can see other functions that in `Utils` class have `is_` prefix if it is validating something.
Please check.

Comment (babenek) [reply credsweeper/utils/util.py]: done ``is_ascii_entropy_validate``

Comment (csh519) [reply credsweeper/utils/util.py]: Thank you.

**Review by csh519 — APPROVED** [review]
Thank you for adding JIRA/Conf. PAT rule and other improvements.

LGTM :+1:

**Review by kmnls — APPROVED** [review]

**Pull request merged**